Form/ChoosePaymentMethodType.php

```diff
@@ -92,15 +92,18 @@ protected function buildChoiceList(FormBuilderInterface $builder, array $options
             $label = 'form.label.'.$method;
 
             if (Legacy::formChoicesAsValues()) {
-                $options['choices'][$method] = $label;
-            } else {
                 $options['choices'][$label] = $method;
+                if (Legacy::needsChoicesAsValuesOption()) {
+                    $options['choices_as_values'] = true;
```

Comment:
I haven't had much bandwidth to look into this. I'll review as soon as I have some free time.